pkg/puller/puller.go

```diff
@@ -151,6 +151,12 @@ func (p *Puller) manage() {
 						peerEntry := peer{addr: peerAddr, po: po}
 						peersToRecalc = append(peersToRecalc, peerEntry)
 					}
+				} else {
```

Comment:
Yep, that will the no-longer-in-neighborhood peers to recalc and be cancelled.  But they will now be queued to cancel every time through the loop because they're never removed from p.syncPeers which is where bp comes from.

I had this code for a while with a logger message and couldn't figure out why my bin1 peer was continually recalcing even though the depth was 13.  I finally changed this else to:

				} else {
					if _, ok := bp[peerAddr.String()]; ok {	// Disconnect syncing peers no longer within depth
						// already syncing, recalc
						peerEntry := peer{addr: peerAddr, po: po}
						peersDisconnected[peerAddr.String()] = peerEntry
						p.logger.Debugf("puller cancelling peer %s po %d depth %d", peerAddr, po, depth)
					}
				}